hash.c

```diff
@@ -303,6 +303,18 @@ objid_hash(VALUE obj)
  *
  * Certain core classes such as Integer use built-in hash calculations and
  * do not call the #hash method when used as a hash key.
+ *
+ * When implementing your own #hash based on multiple values, the best
+ * practice is to combine values using the hash code of an array:
```

Comment:
Should the text also mention the class? (now I agree it's good to include it for most cases)
Might also want to have a note somewhere explaining why having the class there is useful, I think it's non-trivial for most people.

Reply:
Similarly, I was wondering if it might make sense to mention including the same attributes in hashing and equality.

Reply:
I'm happy to merge this PR if my comment just above is addressed.

Reply:
Done. I don't know what's going on with CI though - is `Compilations / ${{ matrix.entry.name }} (pull_request)` supposed to be a job name?

Reply:
I think that's just because it's not expanded due to being skipped as this is a doc-only change.
Not sure how it finds out this is a doc-only change though.

Reply:
I think https://github.com/ruby/ruby/blob/3a8d60f50388016a677f250eb4d7c8bbe9d75d7d/.github/workflows/compilers.yml#L5-L13 controls skipping for that particular job